web_src/js/components/ActivityHeatmap.vue

```diff
@@ -74,6 +64,14 @@ export default {
 
       const newSearch = params.toString();
       window.location.search = newSearch.length ? `?${newSearch}` : '';
+    },
+    tooltipFormatter(v, locale) {
+      const number = v.count.toLocaleString();
+      const datetime = v.date.toISOString();
+      const fallback = v.date.toLocaleDateString();
+      const date = `<relative-time format="datetime" year="numeric" month="short" day="numeric" weekday="" datetime="${datetime}">${fallback}</relative-time>`;
+      const stringToFormat = v.count === 1 ? locale.contributions_on_1 : locale.contributions_on_n;
```

Comment:
I would really suggest not to do the `TrN` on frontend again. Backend doesn't simply use `n === 1`, but:

![image](https://user-images.githubusercontent.com/2114189/232199242-d0cfe784-9e4e-44d4-b10d-63dcac899e33.png)

Reply:
I guess we had better really to wait for a complete ICU message support, instead of making the problem too complex, which would make the migrating to ICU message more difficult .....

Reply:
Would you prefer this PR without the tooltip localization?

Reply:
Yup, if not that complex, I think it looks good.

Reply:
done https://github.com/go-gitea/gitea/pull/24131/commits/a83761cbbae3c2e3b4bced71e680f44432073ac8